app/views/campaign_pages/edit.slim

```diff
@@ -0,0 +1,13 @@
+= form_for(@campaign_page, class: 'form-horizontal') do |f|
+  h2 = @campaign_page.title
+
+  div.form-group
+    = f.label :title, class: 'control-label'
+    = f.text_field :title, class: 'col-lg-3', placeholder: @campaign_page.title
+
+  - @campaign_page.campaign_pages_widgets.each do |widget|
```

Comment:
Yes. Widget_type_id is undefined the way widgets are passed around with the changes, because I'm using the same forms for creating and editing - I'm reading that from the DB but that should optimally be moved to some controller, if only possible.